src/axolotl/cli/main.py

```diff
@@ -0,0 +1,322 @@
+
+    def decorator(function):
+        for name, field in reversed(config_class.model_fields.items()):
+            option_name = f"--{name.replace('_', '-')}"
+
+            if field.annotation == bool:
+                function = click.option(
+                    option_name, is_flag=True, default=None, help=field.description
+                )(function)
+            else:
+                function = click.option(
+                    option_name, default=None, help=field.description
+                )(function)
+        return function
+
+    return decorator
+
+
+@cli.command()
+@click.argument("config", type=str)
+@add_options_from_dataclass(PreprocessCliArgs)
+def preprocess(config: str, **kwargs):
+    """Preprocess datasets before training."""
+    kwargs = {k: v for k, v in kwargs.items() if v is not None}
+
+    from axolotl.cli.preprocess import do_cli
+
+    do_cli(config=config, **kwargs)
+
+
+@cli.command()
+@click.argument("config", type=str)
+@click.option(
+    "--accelerate",
+    is_flag=True,
+    default=True,
+    help="Use accelerate launch for multi-GPU training",
+)
+@add_options_from_dataclass(TrainerCliArgs)
+@add_options_from_config(AxolotlInputConfig)
+def train(config: str, accelerate: bool, **kwargs):
+    """Train or fine-tune a model."""
+    kwargs = {k: v for k, v in kwargs.items() if v is not None}
+
+    if accelerate:
+        base_cmd = ["accelerate", "launch", "-m", "axolotl.cli.train"]
+        if config:
+            base_cmd.append(config)
+        cmd = build_command(base_cmd, kwargs)
+        subprocess.run(cmd, check=True)  # nosec B603
+    else:
+        from axolotl.cli.train import do_cli
+
+        do_cli(config=config, **kwargs)
+
+
+@cli.command()
+@click.argument("config", type=str)
+@click.option(
+    "--accelerate",
+    is_flag=True,
+    default=True,
+    help="Use accelerate launch for multi-GPU inference",
+)
+@click.option("--lora-model-dir", help="Directory containing LoRA model")
+@click.option("--base-model", help="Path to base model for non-LoRA models")
+@click.option("--gradio", is_flag=True, help="Launch Gradio interface")
+@click.option("--load-in-8bit", is_flag=True, help="Load model in 8-bit mode")
+@add_options_from_dataclass(TrainerCliArgs)
+@add_options_from_config(AxolotlInputConfig)
+def inference(config: str, accelerate: bool, **kwargs):
+    """Run inference with a trained model."""
+    kwargs = {k: v for k, v in kwargs.items() if v is not None}
+
+    if accelerate:
+        base_cmd = ["accelerate", "launch", "-m", "axolotl.cli.inference"]
+        if config:
+            base_cmd.append(config)
+        cmd = build_command(base_cmd, kwargs)
+        subprocess.run(cmd, check=True)  # nosec B603
+    else:
+        from axolotl.cli.inference import do_cli
+
+        do_cli(config=config, **kwargs)
+
+
+@cli.command()
+@click.argument("config", type=str)
+@click.option(
+    "--accelerate",
+    is_flag=True,
+    default=True,
```

Comment:
let's make the default for this False, I think shard is likely best on single process as it's taking a large model bin/safetensors file and splitting it into sharded files.